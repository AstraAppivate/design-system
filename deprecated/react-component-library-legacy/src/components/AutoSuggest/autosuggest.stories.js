import React from 'react'
import { withKnobs, text, object } from '@storybook/addon-knobs'
import { storiesOf } from '@storybook/react'

import State from '@royalnavy/storybook-react-input-state'

import AutoSuggest from './index'

export const suggestions = [
  {
    label: 'Ashley',
    town: 'Hull',
  },
  {
    label: 'Dixie',
    town: 'Portsmouth',
  },
  {
    label: 'Hayden',
    town: 'Colchester',
  },
  {
    label: 'Chen',
    town: 'London',
  },
  {
    label: 'Lowe',
    town: 'Bognor Regis',
  },
  {
    label: 'Ginger',
    town: 'Hull',
  },
  {
    label: 'Reynolds',
    town: 'London',
  },
  {
    label: 'Kirby',
    town: 'Birmingham',
  },
  {
    label: 'Hester',
    town: 'London',
  },
  {
    label: 'Rutledge',
    town: 'London',
  },
  {
    label: 'Rosalinda',
    town: 'Ipswich',
  },
  {
    label: 'Tisha',
    town: 'Liverpool',
  },
  {
    label: 'Mcgee',
    town: 'Guildford',
  },
  {
    label: 'Rowe',
    town: 'London',
  },
  {
    label: 'Aida',
    town: 'Guildford',
  },
  {
    label: 'Lane',
    town: 'Manchester',
  },
  {
    label: 'Shelly',
    town: 'York',
  },
  {
    label: 'Jeannine',
    town: 'Edinburgh',
  },
  {
    label: 'Allen',
    town: 'Manchester',
  },
  {
    label: 'Enid',
    town: 'Bath',
  },
  {
    label: 'Duncan',
    town: 'Reading',
  },
  {
    label: 'Pruitt',
    town: 'Cardiff',
  },
  {
    label: 'Marshall',
    town: 'Belfast',
  },
  {
    label: 'Hewitt',
    town: 'Doncaster',
  },
  {
    label: 'Gloria',
    town: 'Hull',
  },
  {
    label: 'Guadalupe',
    town: 'Bath',
  },
  {
    label: 'Wagner',
    town: 'Birmingham',
  },
  {
    label: 'Madeleine',
    town: 'Southampton',
  },
  {
    label: 'Samantha',
    town: 'Northampton',
  },
  {
    label: 'Freeman',
    town: 'Norwich',
  },
  {
    label: 'Trisha',
    town: 'Brighton',
  },
  {
    label: 'Horton',
    town: 'Exeter',
  },
  {
    label: 'Erin',
    town: 'Sheffield',
  },
  {
    label: 'Bailey',
    town: 'Blackpool',
  },
  {
    label: 'Nona',
    town: 'Dublin',
  },
  {
    label: 'Dunlap',
    town: 'Oxford',
  },
  {
    label: 'Addie',
    town: 'Leeds',
  },
  {
    label: 'Earlene',
    town: 'Cambridge',
  },
  {
    label: 'Eleanor',
    town: 'Aberdeen',
  },
  {
    label: 'Tamara',
    town: 'Harrogate',
  },
  {
    label: 'Katherine',
    town: 'Bounemouth',
  },
  {
    label: 'Mcdaniel',
    town: 'Southend-on-sea',
  },
  {
    label: 'Alfreda',
    town: 'Stoke-on-Trent',
  },
  {
    label: 'Goldie',
    town: 'Canturbury',
  },
  {
    label: 'Morgan',
    town: 'Cheltenham',
  },
  {
    label: 'Gates',
    town: 'Doncaster',
  },
  {
    label: 'Barbara',
    town: 'Swansea',
  },
  {
    label: 'Martina',
    town: 'Bolton',
  },
  {
    label: 'Goff',
    town: 'Whitby',
  },
  {
    label: 'Nadia',
    town: 'Colchester',
  },
]

const stories = storiesOf('AutoSuggest', module)
stories.addDecorator(withKnobs)
stories.addParameters({
  info: {
    text: `This component is an implementation of react-autosuggest,
    see https://www.npmjs.com/package/react-autosuggest
    for full instructions on how to implement it.
    `,
    inline: true,
    header: false,
    propTablesExclude: [State],
  },
})

stories.add('Default', () => (
  <State>
    <AutoSuggest suggestions={object('suggestions', suggestions)} />
  </State>
))

stories.add('Custom placeholder', () => (
  <State>
    <AutoSuggest
      suggestions={suggestions}
      placeholder={text('Placeholder', 'Type "A"')}
    />
  </State>
))

stories.add('Custom filter', () => (
  <State>
    <AutoSuggest
      suggestions={suggestions}
      placeholder={text('Placeholder', 'Filter by town')}
      filter={text('Filter', 'town')}
    />
  </State>
))

stories.add('Custom filter and target', () => (
  <State>
    <AutoSuggest
      suggestions={suggestions}
      filter={text('Filter', 'town')}
      target={text('Target', 'town')}
    />
  </State>
))
