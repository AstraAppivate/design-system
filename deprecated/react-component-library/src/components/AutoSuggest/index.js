import React from 'react'
import PropTypes from 'prop-types'
import Autosuggest from 'react-autosuggest'
import { escape, unescape } from 'lodash'

class AutoSuggest extends React.Component {
  /**
   * This component is an implementation of react-autosuggest,
   * see https://www.npmjs.com/package/react-autosuggest
   * for full instructions on how to implement it.
   */

  static propTypes = {
    /** A list of suggestions in array format */
    suggestions: PropTypes.instanceOf(Array),
    /** If you don't want to filter on the 'label' key, you can specify a custom filter */
    filter: PropTypes.string,
    /** If you do not want to display the value of 'label' you can specify which value you want to render */
    renderedVal: PropTypes.string,
    /** You can specify any placeholder string you like */
    placeholder: PropTypes.string,
  }

  static defaultProps = {
    suggestions: [],
    filter: 'label',
    renderedVal: 'label',
    placeholder: 'Begin typing...',
  }

  constructor(props) {
    super(props)
    this.filter = this.props.filter
    this.suggestions = this.props.suggestions
    this.renderedVal = this.props.renderedVal
    this.getSuggestions = value => {
      const escapedValue = escape(value.trim())

      if (escapedValue === '') {
        return []
      }

      const regex = new RegExp(`^${escapedValue}`, 'i')

      return this.suggestions.filter(suggestion =>
        regex.test(suggestion[this.filter])
      )
    }

    this.getSuggestionValue = suggestion => suggestion[this.filter]

    this.renderSuggestion = suggestion => (
      <span>{suggestion[this.renderedVal]}</span>
    )

    this.state = {
      value: '',
      suggestions: this.getSuggestions(''),
    }
  }

  onChange = (event, { newValue }) => {
    this.setState({
      value: newValue,
    })
  }

  onSuggestionsFetchRequested = ({ value }) => {
    this.setState({
      suggestions: this.getSuggestions(value),
    })
  }

  onSuggestionsClearRequested = () => {
    this.setState({
      suggestions: [],
    })
  }

  render() {
    const { value, suggestions } = this.state
    const { placeholder } = this.props
    const inputProps = {
      placeholder: unescape(placeholder),
      value,
      onChange: this.onChange,
    }

    /* Manually specify classees to use */
    const theme = {
      container: 'rn-autosuggest',
      containerOpen: 'rn-autosuggest open',
      input: 'rn-autosuggest__input',
      inputOpen: 'rn-autosuggest__input open',
      inputFocused: 'rn-autosuggest__input--focused',
      suggestionsContainer: 'rn-autosuggest__suggestions',
      suggestionsContainerOpen: 'rn-autosuggest__suggestions open',
      suggestionsList: 'rn-autosuggest__suggestions-list',
      suggestion: 'rn-autosuggest__suggestion',
      suggestionFirst: 'rn-autosuggest__suggestion--first',
      suggestionHighlighted: 'rn-autosuggest__suggestion--highlighted',
      sectionContainer: 'rn-autosuggest__section-container',
      sectionContainerFirst: 'rn-autosuggest__section-container--first',
      sectionTitle: 'rn-autosuggest__section-title',
    }

    return (
      <Autosuggest
        theme={theme}
        suggestions={suggestions}
        onSuggestionsFetchRequested={this.onSuggestionsFetchRequested}
        onSuggestionsClearRequested={this.onSuggestionsClearRequested}
        getSuggestionValue={this.getSuggestionValue}
        renderSuggestion={this.renderSuggestion}
        inputProps={inputProps}
      />
    )
  }
}

export default AutoSuggest
