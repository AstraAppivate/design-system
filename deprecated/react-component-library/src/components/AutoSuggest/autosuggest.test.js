import React from 'react'
import { shallow } from 'enzyme'

import AutoSuggest from './index'

import { suggestions } from './autosuggest.stories'

describe('AutoSuggest', () => {
  let onChangeStub
  let wrapper

  beforeAll(() => {
    onChangeStub = jest.fn()
  })

  afterEach(() => {
    onChangeStub.mockReset()
  })

  describe('text', () => {
    beforeEach(() => {
      wrapper = shallow(<AutoSuggest suggestions={suggestions} />)
    })

    it('should render correctly render the AutoSuggest component', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})
