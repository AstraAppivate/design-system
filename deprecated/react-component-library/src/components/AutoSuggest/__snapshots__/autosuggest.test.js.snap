// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutoSuggest text should render correctly render the AutoSuggest component 1`] = `
<Autosuggest
  alwaysRenderSuggestions={false}
  focusInputOnSuggestionClick={true}
  getSuggestionValue={[Function]}
  highlightFirstSuggestion={false}
  id="1"
  inputProps={
    Object {
      "onChange": [Function],
      "placeholder": "Begin typing...",
      "value": "",
    }
  }
  multiSection={false}
  onSuggestionsClearRequested={[Function]}
  onSuggestionsFetchRequested={[Function]}
  renderSuggestion={[Function]}
  renderSuggestionsContainer={[Function]}
  shouldRenderSuggestions={[Function]}
  suggestions={Array []}
  theme={
    Object {
      "container": "rn-autosuggest",
      "containerOpen": "rn-autosuggest open",
      "input": "rn-autosuggest__input",
      "inputFocused": "rn-autosuggest__input--focused",
      "inputOpen": "rn-autosuggest__input open",
      "sectionContainer": "rn-autosuggest__section-container",
      "sectionContainerFirst": "rn-autosuggest__section-container--first",
      "sectionTitle": "rn-autosuggest__section-title",
      "suggestion": "rn-autosuggest__suggestion",
      "suggestionFirst": "rn-autosuggest__suggestion--first",
      "suggestionHighlighted": "rn-autosuggest__suggestion--highlighted",
      "suggestionsContainer": "rn-autosuggest__suggestions",
      "suggestionsContainerOpen": "rn-autosuggest__suggestions open",
      "suggestionsList": "rn-autosuggest__suggestions-list",
    }
  }
/>
`;
